pandas/core/indexes/multi.py

```diff
@@ -3454,12 +3454,14 @@ def equals(self, other: object) -> bool:
 
         for i in range(self.nlevels):
             self_codes = self.codes[i]
+            other_codes = other.codes[i]
+            if not np.array_equal(self_codes == -1, other_codes == -1):
```

Comment:
prob could define
```
self_mask = self_codes ==  -1
other_mask = other_codes == -1
```
and use here and L3460 and L3465

Reply:
Good point, is probably faster.

Thx